packages/kobotoolbox/src/Adaptor.js

```diff
@@ -41,78 +41,83 @@ export function execute(...operations) {
 
-    const response = await util.request(state, 'GET', url, resolvedOptions);
-    console.log('✓', response.body.count, 'forms fetched.');
-    return util.prepareNextState(state, response, callback);
+    console.log('✓', response.body.results.length, 'forms fetched.');
+    return util.prepareNextState(state, response);
   };
 }
 
 /**
  * Get submissions for a specific form
- * @example
- * getSubmissions({formId: 'aXecHjmbATuF6iGFmvBLBX'}, state => {
- *   console.log(state.data);
- *   return state;
- * });
+ * @example <caption>Get all submissions for a specific form</caption>
+ * getSubmissions('aXecHjmbATuF6iGFmvBLBX');
+ * @example <caption>Get form submissions with a query</caption>
+ * getSubmissions('aXecHjmbATuF6iGFmvBLBX', { query: { _submission_time:{ $gte: "2022-06-12T21:54:20" } } });
  * @function
  * @public
- * @param {object} params - Form Id and data to make the fetch or filter
- * @param {function} callback - (Optional) Callback function to execute after fetching form submissions
+ * @param {string} formId - Form Id to get the specific submissions
+ * @param {object} [options={}] - Optional query params for the request
  * @returns {Operation}
  */
-export function getSubmissions(params, callback) {
+export function getSubmissions(formId, options = {}) {
   return async state => {
-    const [resolvedParams] = expandReferences(state, params);
-
-    const { formId } = resolvedParams;
-
-    const url = `/assets/${formId}/data/?format=json`;
-
-    const response = await util.request(state, 'GET', url, resolvedParams);
-    console.log('✓', response.body.count, 'forms fetched.');
-    return util.prepareNextState(state, response, callback);
+    const [resolvedFormId, resolvedOptions] = expandReferences(
+      state,
+      formId,
+      options
+    );
+
+    const url = `/assets/${resolvedFormId}/data/`;
+    const query = {}
+    if (resolvedOptions.query) {
+      if (typeof resolvedOptions.query == 'string') {
+        query.query = resolvedOptions.query
+      } else {
+        query.query = JSON.stringify(resolvedOptions.query)
+      }
+    }
+
+    const response = await util.request(state, 'GET', url, {
+      paginate: true,
+      query,
+    });
+    console.log('✓', response.results.length, 'forms fetched.');
+    return util.prepareNextState(state, response);
   };
 }
 
 /**
  * Get deployment information for a specific form
  * @example
- * getDeploymentInfo({formId: 'aXecHjmbATuF6iGFmvBLBX'}, state => {
- *   console.log(state.data);
- *   return state;
- * });
+ * getDeploymentInfo('aXecHjmbATuF6iGFmvBLBX');
  * @function
  * @public
- * @param {object} params - Form Id and data to make the fetch or filter
- * @param {function} callback - (Optional) Callback function to execute after fetching form deployment information
+ * @param {string} formId - Form Id to get the deployment information
+ * @param {RequestOptions} [options={}] - Optional query params and headers for the request
```

Comment:
I don't think we need any options on deployments? The spec says there should be no options, and looking a kobo docs I still agree that that's correct